driver/main.c

```diff
@@ -1217,8 +1218,9 @@ static int ppm_mmap(struct file *filp, struct vm_area_struct *vma)
 
 			pfn = vmalloc_to_pfn(vmalloc_area_ptr);
 
+			pgprot_val(vma->vm_page_prot) = pgprot_val(PAGE_SHARED) | _PAGE_ENC;
```

Comment:
I am not sure _PAGE_ENC is all we need here...

Reply:
In theory, _PAGE_ENC should be: https://elixir.bootlin.com/linux/latest/source/arch/x86/include/asm/pgtable_types.h#L153

Reply:
Something similar was made in https://lwn.net/Articles/103656/ (but with other flags, no `_PAGE_ENC` is used)